kernel/rros/clock.rs

```diff
@@ -586,13 +587,91 @@ fn set_timer_value(timer: Arc<SpinLock<RrosTimer>>, value: &Itimerspec64) -> Res
     return Ok(0);
 }
 
+pub fn rros_current_rq() -> *mut rros_rq {
+    let current = rros_current();
+    unsafe { (*(*current).locked_data().get()).rq.unwrap() }
+}
+
+pub fn double_timer_base_lock(tb1: *mut RrosTimerbase, tb2: *mut RrosTimerbase) {
+    unsafe {
+        if tb1 == tb2 {
+            (*tb1).lock.raw_spin_lock();
+        } else if tb1 < tb2 {
+            (*tb1).lock.raw_spin_lock();
+            (*tb2).lock.raw_spin_lock_nested(bindings::SINGLE_DEPTH_NESTING);
+        } else {
+            (*tb2).lock.raw_spin_lock();
+            (*tb1).lock.raw_spin_lock_nested(bindings::SINGLE_DEPTH_NESTING);
+        }    
+    }
+}
+
+pub fn double_timer_base_unlock(tb1: *mut RrosTimerbase, tb2: *mut RrosTimerbase) {
+    unsafe {
+        (*tb1).lock.raw_spin_unlock();
+        if tb1 != tb2 {
+            (*tb2).lock.raw_spin_unlock();
+        }
+    }
+}
+
+pub fn rros_move_timer(timer: Arc<SpinLock<RrosTimer>>, clock: *mut RrosClock, mut rq: *mut rros_rq) {
+    let cpu = get_clock_cpu(unsafe { &(*(*clock).get_master()) }, rros_rq_cpu(rq));
```

Comment:
We often use `master` instead of `get_master` in API. But it is not a big problem here. 
FYI: https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter

Reply:
OK, I'll add a **TODO:** in the code. We can change it in the future when we refactor.